packages/medusa/src/interfaces/payment-service.ts

```diff
@@ -76,7 +107,31 @@ export abstract class AbstractPaymentService
     data: Data
   ): Promise<PaymentSessionData>
 
+  /**
+   * @param context The type of this argument is meant to be temporary and once the previous method signature
+   * will be removed, the type will only be PaymentContext instead of Cart & PaymentContext
+   */
+  public abstract createPayment(
+    context: Cart & PaymentContext
+  ): Promise<PaymentSessionResponse>
+
+  /**
+   * @deprecated use createPayment(context: CreateSessionContext): Promise<PaymentSessionResponse> instead
+   * @param cart
+   */
   public abstract createPayment(cart: Cart): Promise<PaymentSessionData>
+
+  /**
+   * @param context
+   */
+  public abstract createPaymentNew(
+    context: PaymentContext
```

Comment:
thought: can we kill all the `*New` methods?

Reply:
included [here](https://github.com/medusajs/medusa/pull/2743/commits/5b30e2016c3ff51e2dc482d78f2f9f975a115326)